shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/BridgeIngressServiceImpl.java

```diff
@@ -30,14 +33,48 @@
 
     @Override
     public void createBridgeIngress(BridgeDTO bridgeDTO) {
-        final Namespace namespace = customerNamespaceProvider.fetchOrCreateCustomerNamespace(bridgeDTO.getCustomerId());
+        final Namespace namespace = customerNamespaceProvider.getOrCreateCustomerNamespace(bridgeDTO.getCustomerId());
 
         kubernetesClient
                 .resources(BridgeIngress.class)
                 .inNamespace(namespace.getMetadata().getName())
                 .create(BridgeIngress.fromDTO(bridgeDTO, namespace.getMetadata().getName(), ingressImage));
     }
 
+    @Override
+    public Deployment getOrCreateBridgeIngressDeployment(BridgeIngress bridgeIngress) {
+        Deployment deployment = kubernetesClient.apps().deployments().inNamespace(bridgeIngress.getMetadata().getNamespace()).withName(bridgeIngress.getMetadata().getName()).get();
+
+        if (deployment != null) {
+            return deployment;
+        }
+
+        deployment = TemplatesUtils.loadIngressDeploymentTemplate();
```

Comment:
Instead of util, maybe you could consider a real provider/service here.

Reply:
Moved to `TemplateProvider`. The logic is that `TemplateProvider` provides just the template, then the `BridgeIngressService` populates it